src/d/actor/d_a_bst.cpp

```diff
@@ -4,213 +4,2399 @@
+                                            hand[0]->field_0x10D6 = 10;
+                                            hand[0]->mDamage = 0;
+                                        }
+                                    } else {
+                                        if (hand[1]->field_0x10D6 < 5 && hand[1]->mStateTimer == 0) {
+                                            hand[1]->field_0x10D6 = 10;
+                                            hand[1]->mDamage = 0;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+
+                        s16 unk;
+                        if (i_this->health == 0) {
+                            unk = cM_rndF(30.0f) + 30.0f;
+                        } else if (i_this->health == 1) {
+                            unk = cM_rndF(40.0f) + 50.0f;
+                        } else {
+                            unk = cM_rndF(50.0f) + 70.0f;
+                        }
+                        i_this->field_0x2E7E[0] = unk;
+                    }
+
+                    boss->field_0x2E74[1] = 10;
+                    boss->field_0x2E74[0] = 10;
+                    if (i_this->field_0x2E7E[2] == 0 && hand[0]->field_0x10D6 == 6 && hand[1]->field_0x10D6 == 6) {
+                        i_this->field_0x2E7C = 12;
+                        hand[0]->field_0x10FC[1] = 100;
+                        hand[1]->field_0x10FC[1] = 100;
+
+                        fopAcM_seStart(boss, JA_SE_CM_BST_HEAD_EYE_OPEN, 0);
+                        mDoAud_bgmStart(JA_BGM_UNK_122);
+
+                        i_this->field_0x2E7E[1] = cM_rndF(50.0f) + 150.0f;
+                    }
+                }
+                break;
+            case 12:
+                if (i_this->field_0x2E7E[1] == 0 && boss->field_0x10D6 == 1) {
+                    boss->field_0x10D6 = 14;
+                    boss->mDamage = 0;
+                    boss->field_0x10D0 = REG0_F(7) + (-500.0f);
+                }
+                hand[0]->field_0x10D6 = 6;
+                hand[0]->field_0x10FC[1] = 3;
+                hand[1]->field_0x10FC[1] = 3;
+                if (boss->field_0x10D6 == 7 && boss->mDamage >= 7) {
+                    i_this->field_0x2E7C = 11;
+                    i_this->field_0x2E7E[2] = 50;
+                }
+                break;
+        }
+    }
 }
 
 /* 00006DC0-00006FA4       .text beam_eff_set__FP4cXyzsUc */
-void beam_eff_set(cXyz*, short, unsigned char) {
-    /* Nonmatching */
+static void beam_eff_set(cXyz* pos, short y, unsigned char set_angle) {
+    if (!set_angle) {
+        dComIfGp_particle_set(dPa_name::ID_SCENE_81BF, pos);
+        dComIfGp_particle_set(dPa_name::ID_SCENE_A1C0, pos);
+        dComIfGp_particle_set(dPa_name::ID_SCENE_A1C1, pos);
+    } else {
+        // x is -2.0
+        csXyz angle(-0x4000, y, 0);
```

Comment:
What does this comment mean?

Reply:
Not sure! Must have been a note for myself. Removed.